### ISSUE TITLE
Fix TravisCI build

### ISSUE THREAD

**Issue by tdauth** [tdauth/wc3lib#9]
If it does not work, just use Docker like in the repository https://github.com/tdauth/cpp-futures-promises.

**Comment by Karaulov** [comment]
There are very many dependencies and requirements, it will be hard to compile for Windows manually. Please add an automatic build for Windows.

**Comment by tdauth** [comment]
Can you provide some changes for the TravisCI file to build it on Windows? I have never build a project on TravisCI on Windows.

**Comment by tdauth** [comment]
You can try to use https://github.com/ruslo/hunter
If you modify the CMakeLists.txt files and all packages with Hunter, it might build on Windows.
Otherwise, you have to use a VM with Linux.

**Comment by Karaulov** [comment]
wc3lib ever build for Windows? Or only "theoretically" supports Windows?

**Comment by tdauth** [comment]
I have built it without the MDLX option and a cygwin compiler. I had to built Boost manually etc. It's very hard to achieve, so an automatic build with TravisCI would be good but TravisCI does not support Windows?

**Comment by Karaulov** [comment]
Sorry, I do not know, never used it.

**Comment by Karaulov** [comment]
https://www.appveyor.com/docs/build-environment/ 

This cannot be used for auto build ?

**Comment by tdauth** [comment]
Isn't it commercial? How can I use it with GitHub?

**Comment by Karaulov** [comment]
I don't know, i use it for build Visual Studio project for free.

**Comment by tdauth** [comment]
You can build without certain CMake options:
* Disable the option USE_KIOSLAVE
* Disable the option MDLX
* Disable the option USE_OGREBLP

Then you should not require OGRE nor KF5 KIO anymore. Only Qt5.

**Comment by Karaulov** [comment]
I found ogre for windows, but can't find KDE error:
include could not find load file:

    KDEInstallDirs

**Comment by Karaulov** [comment]
OGRE and KF5 KIO not needed for wc3lib completed build ?

**Comment by tdauth** [comment]
No, they are not. All dependencies can be configured with the options.

**Comment by Karaulov** [comment]
I try to build on my PC.

Got 100000 errors in mapstrings.cpp file. 

BOOST 1.63.0 version.


if problem in 1.63 version, appveyor also not support 1.61 version: 
![xsttbwb 1](https://user-images.githubusercontent.com/6445336/36305393-69f342bc-1324-11e8-813c-a4e6713c252c.png)

**Comment by tdauth** [comment]
Can you list some of the errors?

**Comment by Karaulov** [comment]
Yes, but russian :)

```
4>mapstrings.cpp
4>C:\BOOST\boost/proto/matches.hpp(835): error C2039: proto_tag: не является членом "boost::spirit::attribute<0>"
4>C:\BOOST\boost/spirit/home/support/context.hpp(235): note:  см. объявление "boost::spirit::attribute<0>"
4>C:\BOOST\boost/phoenix/core/meta_grammar.hpp(66): note: выполняется компиляция ссылки на экземпляр шаблон класс "boost::proto::switch_<boost::phoenix::meta_grammar,boost::proto::tag_of<boost::proto::_> (void)>::impl<Expr,boost::mpl::bool_<true> &,boost::phoenix::is_nullary &>"
4>        with
4>        [
4>            Expr=const boost::spirit::attribute<0> &
4>        ]
4>C:\BOOST\boost/phoenix/core/is_nullary.hpp(123): note: выполняется компиляция ссылки на экземпляр шаблон класс "boost::phoenix::evaluator::impl<const Expr &,boost::phoenix::vector2<boost::mpl::true_,boost::phoenix::is_nullary>,boost::proto::envns_::empty_env>"
4>        with
4>        [
4>            Expr=boost::spirit::attribute<0>
4>        ]
4>C:\BOOST\boost/phoenix/core/actor.hpp(174): note: выполняется компиляция ссылки на экземпляр шаблон класс "boost::phoenix::result_of::is_nullary<Expr,void>"
4>        with
4>        [
4>            Expr=boost::spirit::attribute<0>
4>        ]
4>C:\BOOST\boost/proto/matches.hpp(830): note: выполняется компиляция ссылки на экземпляр шаблон класс "boost::phoenix::result_of::actor<T>"
4>        with
4>        [
4>            T=boost::spirit::attribute<0>
4>        ]
4>C:\BOOST\boost/phoenix/core/meta_grammar.hpp(42): note: выполняется компиляция ссылки на экземпляр шаблон класс "boost::proto::switch_<boost::phoenix::meta_grammar,boost::proto::tag_of<boost::proto::_> (void)>::impl<Expr,State,Data>"
4>        with
4>        [
4>            Expr=const boost::phoenix::result_of::actor<boost::spirit::attribute<0>>,
4>            State=boost::mpl::bool_<true> &,
4>            Data=boost::phoenix::is_nullary &
4>        ]
4>C:\BOOST\boost/phoenix/core/is_nullary.hpp(169): note: выполняется компиляция ссылки на экземпляр шаблон класс "boost::phoenix::evaluator::impl<const boost::phoenix::result_of::actor<T>,State,Data>"
4>        with
4>        [
4>            T=boost::spirit::attribute<0>,
4>            State=boost::mpl::bool_<true> &,
4>            Data=boost::phoenix::is_nullary &
4>        ]
```

**Comment by tdauth** [comment]
Can you change to English please :D

**Comment by Karaulov** [comment]
I don't know, possible error in mapstrings.cpp(112).



And error in void MpqEditor::addFiles()


const uint64 dataSize = std::streamoff(endPosition(in)) + 1;
byte data[dataSize];

:D :D

**Comment by Karaulov** [comment]
**wc3lib was created only for linux?**

**Comment by tdauth** [comment]
Well, I have compiled it on Windows only once. It's easier to use it on Linux.

**Comment by Karaulov** [comment]
And what is there? Does the editor work on Windows? 

I can't compile JASS, MDLX , without MDLX i can't compile many other projects in solution.

**Comment by Karaulov** [comment]
For editor need build KDE for Windows ?

**Comment by tdauth** [comment]
The editor module supports tools for editting triggers and object data. There is a model viewer as well. The project is quite unfinished. What exactly do you want to do?

**Comment by tdauth** [comment]
Use GitHub actions: https://github.com/marketplace/actions/setup-cpp-c-c
